File: linechart/src/main/java/com/davidmiguel/linechart/formatter/DefaultYAxisValueFormatter.java
package com.davidmiguel.linechart.formatter;

import android.graphics.RectF;

import androidx.annotation.NonNull;

import java.text.DecimalFormat;
import java.text.NumberFormat;

/**
 * Formats the value with two decimals if needed.
 */
public class DefaultYAxisValueFormatter implements YAxisValueFormatter {

    private NumberFormat formatter;

    public DefaultYAxisValueFormatter() {
        formatter =  new DecimalFormat( "#.##");
    }

    @NonNull
    @Override
    public String getFormattedValue(float value, @NonNull RectF dataBounds, int gridYDivisions) {
        return formatter.format(value);
    }
}

File: linechart/src/main/java/com/davidmiguel/linechart/formatter/DefaultYAxisValueFormatter.kt
package com.davidmiguel.linechart.formatter

import android.graphics.RectF
import java.text.DecimalFormat
import java.text.NumberFormat

/**
 * Formats the value with two decimals if needed.
 */
class DefaultYAxisValueFormatter : YAxisValueFormatter {

    private val formatter: NumberFormat = DecimalFormat("#.##")

    override fun getFormattedValue(value: Float, dataBounds: RectF, gridYDivisions: Int): String {
        return formatter.format(value.toDouble()).replace("^-(?=0(,0*)?$)".toRegex(), "") // Remove - sign if 0
    }
}
